core/src/main/scala/scalaz/std/String.scala

```diff
@@ -38,33 +38,32 @@ trait StringFunctions {
   def charsNel(s:String, e: => NonEmptyList[Char]) : NonEmptyList[Char] = charsNel(s) getOrElse e
 
   // Parsing functions.
+  private def intChars(s: String): Boolean = s.forall(c => c.isDigit || c == '-')
+  private def floatChars(s: String): Boolean = s.forall(c => c.isDigit || c == '-' || c == '.')
 
-  def parseBoolean(s:String): Validation[IllegalArgumentException, Boolean] =
-    Validation.fromTryCatchThrowable[Boolean, IllegalArgumentException](s.toBoolean)
+  private def asNumber[T](f: String => T, b: String => Boolean, r: Tuple2[T, T], s: String, t: String): Validation[String, T] =
```

Comment:
you can use `ClassTag` instead of `t: String` and get the primitive name expanded for free.

Reply:
Ah, yes, I was trying `TypeTag` and that wasn't making me happy...

Reply:
Generally there is need for a new `Describable` typeclass with a macro that generates the refined type name. But `ClassTag` is the closest thing we have from stdlib.